components/raftstore/src/store/peer.rs

```diff
@@ -3575,6 +3606,11 @@ pub trait RequestInspector {
             return Ok(RequestPolicy::ProposeNormal);
         }
 
+        let flags = WriteBatchFlags::from_bits_truncate(req.get_header().get_flags());
```

Comment:
Probably should panic on unrecognized flags.

Reply:
I think this could break forward compatibility.

Reply:
It's on purpose. New features should be enabled when all binaries are upgraded. Otherwise different instances can behave differently.

Reply:
The flags of read request are set by tikv itself, not from the network, but I add the check for robustness though.